apps/showcase/src/components/training/training.component.ts

```diff
@@ -0,0 +1,238 @@
+import {firstValueFrom} from 'rxjs';
+import {CopyTextPresComponent} from '../utilities/copy-text';
+import {EditorMode, TrainingProject} from './code-editor-view';
+import {TrainingStepPresComponent} from './training-step';
+
+/** Description of training step */
+interface TrainingStepDescription {
+  /** Step title */
+  title: string;
+  /** URL to step instructions (HTML content) */
+  htmlContentUrl: string;
+  /** Step files configuration */
+  filesConfiguration?: {
+    /** Name of directory in which files will be put */
+    name: string;
+    /** Starting file to be displayed in the project */
+    startingFile: string;
+    /** Commands to run in the project */
+    commands: string[];
+    /** URLs of step project */
+    urls: { [key: string]: string };
+    /** URLs of step solution project */
+    solutionUrls?: { [key: string]: string };
+    /** Mode of the Code Editor */
+    mode: EditorMode;
+  };
+}
+
+/** Training step - consists of its description and its dynamic content */
+type TrainingStep = {
+  /** Description of the training step */
+  description: TrainingStepDescription;
+  /** Dynamic content of the training step */
+  dynamicContent: {
+    /** HTML Content - corresponds to the step instructions */
+    htmlContent: WritableSignal<string>;
+    /** Exercise project of the step */
+    project: WritableSignal<TrainingProject | null>;
+    /** Solution of the exercise project of the step */
+    solutionProject: WritableSignal<TrainingProject | null>;
+  };
+};
+
+/** Resources to get file content */
+type Resource = {
+  /** Resource path */
+  path: string;
+  /** Resource content */
+  content: string;
+};
+
+@Component({
+  selector: 'o3r-training',
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    CopyTextPresComponent,
+    DynamicContentModule,
+    FormsModule,
+    JsonPipe,
+    NgbAccordionModule,
+    NgComponentOutlet,
+    TrainingStepPresComponent,
+    NgbDropdown,
+    NgbDropdownToggle,
+    NgbDropdownMenu,
+    NgbDropdownItem
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './training.component.html',
+  styleUrl: './training.component.scss'
+})
+export class TrainingComponent implements OnInit {
+  /**
+   * Index of the training step to display
+   */
+  public currentStepIndex = 0;
+  public showInstructions = true;
+  public showSolution = false;
+  /**
+   * Training steps that
+   */
+  public steps: WritableSignal<TrainingStep[]> = signal([]);
+  public stepNames = computed(() =>
+    this.steps().map((step) => step.description.title)
+  );
+  private readonly dynamicContentService = inject(DynamicContentService);
+
+  @Input()
+  public trainingPath = '';
+  @Input()
+  public title = '';
+
+  private async loadStepContent(step: TrainingStep) {
+    if (!step.dynamicContent.htmlContent()) {
+      void this.loadResource(step.description.htmlContentUrl).then((content) =>
+        step.dynamicContent.htmlContent.set(content)
+      );
```

Comment:
```suggestion
      const content = await this.loadResource(step.description.htmlContentUrl)
      step.dynamicContent.htmlContent.set(content);
```

Reply:
Yeah, it used to be in ngOnInit/constructor